include/zephyr/drivers/rtc.h

```diff
@@ -0,0 +1,454 @@
+ * enable. If the mask parameter is 0, the alarm will be disabled. The RTC
+ * alarm will trigger when all enabled fields of the alarm time match the RTC
+ * time.
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm
+ * @param mask Mask of fields in the alarm time to enable
+ * @param timeptr The alarm time to set
+ *
+ * @note The timeptr param may be NULL if the mask param is 0
+ *
+ * @note Only the enabled fields in the timeptr param need to be set
+ *
+ * @note Bits in the mask param are defined in the RTC_ALARM_TIME_MASK defines
+ *
+ * @return 0 if successful
+ * @return -EINVAL if id is out of range or time is invalid
+ * @return -ENOSYS if API is not implemented
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+				 const struct tm *timeptr);
+
+static inline int z_impl_rtc_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+					    const struct tm *timeptr)
+{
+	const struct rtc_driver_api *api =
+		(const struct rtc_driver_api *)dev->api;
+
+	return api->alarm_set_time(dev, id, mask, timeptr);
+}
+
+/**
+ * @brief API for getting RTC alarm time
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm
+ * @param mask Destination for mask of fields which are enabled in the alarm time
+ * @param timeptr Destination for the alarm time
+ *
+ * @note bits in mask are stored in RTC_ALARM_TIME_MASK defines
+ *
+ * @return 0 if successful
+ * @return -EINVAL if id is out of range
+ * @return -ENOSYS if API is not implemented
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *mask,
+				 struct tm *timeptr);
+
+static inline int z_impl_rtc_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *mask,
+					    struct tm *timeptr)
+{
+	const struct rtc_driver_api *api =
+		(const struct rtc_driver_api *)dev->api;
+
+	return api->alarm_get_time(dev, id, mask, timeptr);
+}
+
+/**
+ * @brief API for testing if RTC alarm is pending
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm to test
+ *
+ * @return 1 if alarm is pending
+ * @return 0 if alarm is not pending
+ * @return -EINVAL if id is out of range
+ * @return -ENOSYS if API is not implemented
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_alarm_is_pending(const struct device *dev, uint16_t id);
+
+static inline int z_impl_rtc_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+	const struct rtc_driver_api *api =
+		(const struct rtc_driver_api *)dev->api;
+
+	return api->alarm_is_pending(dev, id);
+}
+
+/**
+ * @brief API for clearing RTC alarm pending status
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm for which the pending status will be cleared
```

Comment:
Is there a use case for reading pending status without clearing it?

Reply:
I can not think of one, I will remove the clear_pending API and update the description of the is_pending to clear the pending state when read